src/main/io/statusindicator.c

```diff
@@ -60,20 +76,32 @@ void warningLedBeeper(bool on)
 void warningLedPulse(void)
 {
 	// FIXME this is not really pulse now - original code pulsed the light once a second (on short, off long), this code is a result of rebasing onto commit beac0a35cec690f33a2ef5d13f41f3e3a9d8e57a
-	nextBlinkMask = 0b10000000;
+	nextBlinkMask = 0b1000001;
 }
 
 void warningLedSetBlinkMask(uint32_t newBlinkMask)
 {
 	nextBlinkMask = newBlinkMask;
 }
 
+void warningLedCode(uint8_t code)
+{
+    nextBlinkMask = 0b100000;
+ 	int bitsRemaining = 32 - 6;  // must be an even number
+    for (int i = 0; i < code && bitsRemaining; i++) {
```

Comment:
interesting.  My first cut of code is generally, as you can see, stuff that's simple to understand while you're writing it.  I generally leave optimization until after the first cut.  I'll skip this for now as I have the MSP stuff to do first.